### PR TITLE
let web_extension_initialize handle nullpointers

### PR DESCRIPTION
**Pull request by vhdirk**
This fixes the panic in #40, but that was only ever a side effect of something deeper.

### CODE REVIEW

**Comment by antoyo** [comment]
When will the value be `None`?
I assume you want to use the other macro `web_extension_init` when you don't want to have data.

Maybe there's a better way to handle both situation.

Or should we remove the other macro and only keep this one with the `Option<&Variant>`?

**Comment by vhdirk** [comment]
That's just the thing. I do want to use ```user_data```. In #40, I'm using your example verbatim (https://github.com/gtk-rs/webkit2gtk-rs/blob/master/examples/main.rs), but recently, it always ends up being a null pointer and panics.
So, I'd argue that this would be the only macro needed: If you expect no ```user_data```, you can ignore the ```None``` value safely. If you _do_ expect ```user_data```, you can still gracefully terminate rather than panicing in the macro.

**Comment by antoyo** [comment]
Well, first, we need to understand why we get a null pointer when setting the data.
I don't really have time to check that now, so it might take a couple of weeks for me to dig into this issue.

**Comment by vhdirk** [comment]
I'll see if I can get to the bottom of #40 myself, but I probably won't be able to do that soon either.

File: example/src/lib.rs [hunk 1]
```diff
@@ -76,8 +76,9 @@ pub fn web_extension_initialize(extension: &WebExtension) {
 
 web_extension_init_with_data!();
 
-pub fn web_extension_initialize(extension: &WebExtension, user_data: &Variant) {
-    let _string = user_data.get_str();
+pub fn web_extension_initialize(extension: &WebExtension, user_data: Option<&Variant>) {
+    let ud = user_data.unwrap();
+    let _string = ud.get_str();
```
Comment (EPashkin) [inline example/src/lib.rs]: IMHO this better changed to
```
let user_string = user_data.map(|v| v.get_str()).and_then(|s| s.to_owned());
dbg!(user_string);
```
@antoyo Can you test it with and without features?

File: gir-webkit2gtk-webextensions.toml [hunk 2]
```diff
@@ -10,8 +10,14 @@ external_libraries = [
...
    "Gtk",
-   "Soup",
+   "Soup"
 ]
 
+manual = [
+   "JavaScriptCore.GlobalContext",
+]
+
+
```
Comment (EPashkin) [inline gir-webkit2gtk-webextensions.toml]: IMHO all changes in this file unneeded

File: example/src/lib.rs (hunk 1 above)
Comment (antoyo) [reply example/src/lib.rs]: This works in both cases.

**Comment by antoyo** [comment]
@vhdirk If you apply both changes mentioned by @EPashkin, we can merge this PR.
Thanks.

File: example/src/lib.rs (hunk 1 above)
Comment (EPashkin) [reply example/src/lib.rs]: Good, to hear, thanks for checking.
@vhdirk Please update PR.

**Comment by EPashkin** [comment]
@vhdirk Thanks.
@antoyo IMHO this ready for merge, it you agree using custom JavaScriptCore-4.0.gir and current version javascriptcore-rs (as related code unnecessary included)

**Comment by antoyo** [comment]
There are still `[patch]` sections pointing to other repositories.

**Comment by EPashkin** [comment]
Oh, forgot about that, more reason to remove merge with my PR.
@vhdirk please do this.
@antoyo Anyway if you agree with using current javascriptcore and update it to generated version in other time (I can't do it right), then please merge https://github.com/gtk-rs/javascriptcore-rs/pull/27
and I remove patch in my PRs too.

File: example/src/lib.rs [hunk 3]
```diff
@@ -76,8 +76,9 @@ pub fn web_extension_initialize(extension: &WebExtension) {
 
 web_extension_init_with_data!();
 
-pub fn web_extension_initialize(extension: &WebExtension, user_data: &Variant) {
-    let _string = user_data.get_str();
+pub fn web_extension_initialize(extension: &WebExtension, user_data: Option<&Variant>) {
+    let user_string = user_data.map(|v| v.get_str()).and_then(|s| s.to_owned());
```
Comment (EPashkin) [inline example/src/lib.rs]: I forgot do right type check and to_owned applied to `Option<&str>` and not works 😢 
so this better be
```
let user_string: Option<String> = user_data.and_then(Variant::get_str).map(ToOwned::to_owned);
```
including clippy nagging about https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

**Comment by EPashkin** [comment]
👍 Thanks

**Pull request merged**